### PR TITLE
Start returning a response from sender methods

### PR DESCRIPTION
**Pull request by jmhooper**
**Why**: So we can log that response with events in the IdP

This commit also adds a logger class which can be used to capture raw logs of SMS events

### CODE REVIEW

File: lib/telephony/test/sms_sender.rb [hunk 1]
```diff
@@ -2,7 +2,13 @@ module Telephony
   module Test
     class SmsSender
       def send(message:, to:)
-        Message.messages.push(Message.new(body: message, to: to))
+        error = ErrorSimulator.new.error_for_number(to)
+        if error.nil?
+          Message.messages.push(Message.new(body: message, to: to))
+          Response.new(success: true, extra: { request_id: 'fake-message-request-id' })
+        else
+          Response.new(success: false, error: error, extra: { request_id: 'fake-message-request-id' })
```
Comment (zachmargolis) [inline lib/telephony/test/sms_sender.rb]: Looks like rubocop got mad at the line length on this bit

File: lib/telephony/test/voice_sender.rb [hunk 2]
```diff
@@ -2,7 +2,13 @@ module Telephony
   module Test
     class VoiceSender
       def send(message:, to:)
-        Call.calls.push(Call.new(body: message, to: to))
+        error = ErrorSimulator.new.error_for_number(to)
+        if error.nil?
+          Call.calls.push(Call.new(body: message, to: to))
+          Response.new(success: true, extra: { request_id: 'fake-message-request-id' })
+        else
+          Response.new(success: false, error: error, extra: { request_id: 'fake-message-request-id' })
```
Comment (zachmargolis) [inline lib/telephony/test/voice_sender.rb]: Ditto line length

File: lib/telephony/pinpoint/sms_sender.rb [hunk 3]
```diff
@@ -30,27 +30,55 @@ def send(message:, to:)
             },
           },
         )
-        raise_if_error(response.message_response.result[to])
```
Comment (zachmargolis) [inline lib/telephony/pinpoint/sms_sender.rb]: Looks like this used to raise but now it returns an explicit error object? Does what warrant a version bump? Or is that usually done on a separate PR?)

Comment (jmhooper) [reply lib/telephony/pinpoint/sms_sender.rb]: Yeah, I'm gonna bump the version in a different PR
